docs/tutorials/runtime-module-in-own-crate.md

```diff
@@ -0,0 +1,167 @@
+1. Our module can be easily imported into other nodes in the future.
+2. Our module can be published to places like github or crates.io without publishing our entire node in the same place.
+
+We do still need the node template, however, to actually run our module.
+
+## Clone the Module Template Repository
+So if our module isn't going to live in the node template, where _is_ it going to live? If you aren't familiar with library crates or how to use them in other rust projects, it's may be worth reading the [cargo docs](https://doc.rust-lang.org/cargo/guide/creating-a-new-project.html).
+
+While it's perfectly possible to create a new crate from scratch, the fastest way to get going with a new module is to clone the template repository.
+
+```bash
+git clone https://github.com/shawntabrizi/substrate-module-template
+```
+
+You should be able to immediately compile this template module with
+```bash
+cargo build --release
+```
+
+But before we jump straight to including our module in a runtime, let's take a look at the `Cargo.toml` file.
+
+After the standard naming and authorship information, we see a few lines about the `std` feature. Because substrate runtime code targets wasm, all the dependencies we use must be able to compile with `no_std`. Our `Cargo.toml` file begins by telling dependencies to only use their `std` feature when this module also uses its `std` feature. You will need to add each dependency you include here.
+
+```toml
+[features]
+default = ['std']
+std = [
+    'parity-codec/std',
+    'support/std',
+    'system/std',
+]
+```
+Our module will depend on low level features such as `system` and `support` from substrate itself. It will also be depended upon by a substrate-based runtime from above. So some care may be needed to ensure version compatibility on both sides. Substrate does not yet have releases published on crates.io, so instead we'll use git-based versioning.
+If you plan for third parties to use your module, you should develop against the `v1.0` branch as the template demonstrates. It is also acceptable to develop against specific git commit revisions.
+
+```toml
+# This dependency is published on crates.io so just use a version
+[dependencies.parity-codec]
+default-features = false
+features = ['derive']
+version = '3.5'
+
+# These dependencies specify a git branch
+[dependencies.support]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'srml-support'
+branch = 'v1.0'
+
+[dependencies.system]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'srml-system'
+branch = 'v1.0'
+# Could also use a specific revision
+# rev = '<commit hash>'
+```
+
+The final section of the `Cargo.toml` file specifies the dev dependencies, and is not necessary to make the template compile. Remove that final section to confirm this for yourself.
```

Comment:
This section becomes really weird. Rather than have them remove it and get into an error, add it back, etc..., why not simply specify that `dev-dependencies` are those which are simply used by our tests:

https://doc.rust-lang.org/stable/rust-by-example/testing/dev_dependencies.html

Reply:
I like learning by doing. I could just explain it instead of show it, but a user in a hurry could also just decide to take my work for it and not reproduce the error. There won't be much left to this beyond
1. substrate-node-new
2. clone template module
3. insert module in runtime
if we omit portions like this.

Reply:
Anyway, I've edited to de-emphasize that side exercise.